### PR TITLE
Async onMapInit + test changes to support

### PR DESCRIPTION
**Pull request by webbhead777**
resolves: #13

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+import Map from 'Map'
+
+// utility used by many tests for wrapping a Map component
+export const mountOpts = props => ({
+  wrappingComponent: Map,
+  wrappingComponentProps: {
+    allowAsyncMount: false, // this forces wrappingComponent to render children immediately
+    ...props
+  }
+})
```
Comment (webbhead777) [inline src/index.test.js]: should this be named `mapWrapper` or something more specific than `mountOpts`?

Comment (glenselle) [reply src/index.test.js]: We can drop these mount opts now that we've got async tests working right?

File: package.json [hunk 2]
```diff
@@ -71,14 +73,15 @@
...
-    "jest": "~24.9.0",
+    "jest": "^25.1.0",
     "jest-canvas-mock": "~2.2.0",
     "jest-environment-jsdom-global": "~1.2.0",
     "jest-styled-components": "~7.0.0",
     "jsdoc": "~3.6.3",
     "ol": "4.6.5",
     "react": "~16.11.0",
     "react-dom": "~16.11.0",
+    "react-test-renderer": "^16.13.0",
```
Comment (webbhead777) [inline package.json]: drop this

File: src/Basemaps/BingMaps.test.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import React from 'react'
 import { shallow, mount } from 'enzyme'
-import BingMaps from './BingMaps'
+import { cleanup, fireEvent, getByText, render, waitFor } from '@testing-library/react'
```
Comment (webbhead777) [inline src/Basemaps/BingMaps.test.js]: move these destructured to scoped within tests

File: src/Basemaps/BingMaps.test.js [hunk 4]
```diff
@@ -10,29 +12,38 @@ const mockSourceOpts = {
 }
 
 describe('<BingMaps />', () => {
-  it('should render a basic basemap option component', () => {
-    const wrapper = shallow(<BingMaps sourceOpts={mockSourceOpts} />, { wrappingComponent: Map })
+  afterEach(cleanup)
```
Comment (webbhead777) [inline src/Basemaps/BingMaps.test.js]: drop these

File: src/Map/Map.js [hunk 5]
```diff
@@ -66,34 +77,43 @@ class Map extends React.Component {
...
-          fullScreen={fullScreen} />
+          fullScreen={fullScreen}
+          style={style} />
         <MapContext.Provider value={this.getContextValue()}>
-          {!this.map // wait for a map to exist before rendering children that need a ref to map
-            ? null
-            : children}
+          {/* wait for map to initialize before rendering children
+            unless a synchronous mount is forced, render children before map is initialized */
+            mapInitialized || !allowAsyncMount
```
Comment (webbhead777) [inline src/Map/Map.js]: drop `allowAsyncMount`

File: src/Map/utils.test.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
 import { mount } from 'enzyme'
+import { mountOpts } from 'index.test'
```
Comment (webbhead777) [inline src/Map/utils.test.js]: drop this import

File: src/index.test.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
 import * as ol_kit from './index' // eslint-disable-line camelcase
+import Map from 'Map'
```
Comment (webbhead777) [inline src/index.test.js]: drop this

**Pull request merged**